feat: shared application key (#1238)
gravitee-io/gravitee-api-management: gravitee-apim-console-webui/docs/management-configuration-portal.md

For example, you are able to select the Theme to apply on the portal or toggle features like Support or Rating

Some settings are read-only because they are set with environment variables.

## API key configuration

The custom API key mode allows you to set arbitrary values as API keys. This way, API publishers are able to silently migrate to Gravitee.io without changing consumers API key.

> The shared API key mode will make consumers able to reuse the same application key across API subscriptions.

Reviewer: ```suggestion
The shared API key mode will make consumers reuse the same API key across all API subscriptions of this application.
```

> With this mode enabled, consumers will be asked on their first subscription to choose between reusing their key

Reviewer: ```suggestion
With this mode enabled, consumers will be asked on their second subscription to choose between reusing their key
```

across all subscriptions or generate one different API key for each subscription (which is the default mode).

This choice is permanent and consumers will not be able to switch back to one key per subscription for their application.

When disabling the shared API key mode, applications that have already been configured to use a shared key will continue

> to work this way, but consumers will stop being asked to chose between one mode or the other on their first subscription.

Reviewer: ```suggestion
to work this way, but consumers will stop being asked to choose between one mode or the other on their second subscription.
```
